include/dfu/flash_img.h

```diff
@@ -20,6 +20,25 @@ struct flash_img_context {
 	struct stream_flash_ctx stream;
 };
 
+
+/**
+ * @brief Structure for verify flash region integrity
+ *
+ * This is used to pass data and select algorithm to be used to check flash
+ * integrity.
+ *
+ * Note: match vector length is fixed and depends on size from hash algorithm
+ * used to verify flash integrity. The default algorithm is SHA-256 and it is
+ * selected when check_integrity parameter is NULL. This enables use of custom
+ * algorithm when necessary.
+ */
+struct flash_img_check {
+	const uint8_t *match;		/** Match vector data */
+	size_t len;			/** Content len to be compared */
+	int (*check_integrity)(struct flash_img_context *ctx,
```

Comment:
The callback is optional and will be rarely used, in most case users probably will rely on default (SHA256) function, so both the part of structure and code that supports it in `flash_img_check`, could be enclosed in conditional compilation with option added in Kconfig enabling customized integrity check procedure.